react/features/app/middleware.js

```diff
@@ -66,6 +71,42 @@ function _connectionEstablished(store, next, action) {
     return result;
 }
 
+/**
+ * Performs an immediate reload of the application if a "split-brain" scenario
+ * is detected. The detection is based on the assumption that a connection error
+ * with "item-not-found" that happens early on in the call is a signal of
+ * split-brain.
+ *
+ * @param {Store} store - The redux store in which the specified {@code action}
+ * is being dispatched.
+ * @param {Dispatch} next - The redux {@code dispatch} function to dispatch the
+ * specified {@code action} to the specified {@code store}.
+ * @param {Action} action - The redux action {@code CONNECTION_FAILED}
+ * which is being dispatched in the specified {@code store}.
+ * @private
+ * @returns {Object|undefined} The new state that is the result of the reduction
+ * of the specified {@code action} or undefined if a reload is triggered.
+ */
+function _connectionFailed({ dispatch, getState }, next, action) {
+    const { error } = action;
+
+    const { timeEstablished } = getState()['features/base/connection'];
+    const { _immediateReloadThreshold } = getState()['features/base/config'];
+    const reloadThreshold = typeof _immediateReloadThreshold === 'number'
+        ? _immediateReloadThreshold : 1500;
+
+    if (error
+        && error.message === 'item-not-found'
+        && error.details
+        && error.details.shard_changed
```

Comment:
Given there is a specific indicator for this, why do we need to take the time into account? Is there a scenario in which we get `item-not-found` and the error indicates the shard has changed, but we **don't** want to reload immediately?

Reply:
Well, this can happen at any time during the conference. And the scenario is this:
- user A got routed to shardA
- user B got routed to shardB
- split brain is detected and is fixed by moving all conferences to shardA, then userB got routed to the new shard and receives item-not-found with shard_changed. 
During the conference (yesterday's planning meeting there was outage in aws us-east for fe wminutes), a shard can fail and all conferences are moved to new shard, the normal reload was shown with random timeout (we don't want many users joining at the same time) and we all re-connected and continued the meeting. this is also item-not-found with shard_changed.

Reply:
Yes, it is. As I was not sure about the exact value for the time while we were discussing this with Lenny, we came to the conclusion that we need a config to control it.
So if we see it one meet.jit.si and it was over 1500 ms we can then quickly change it through config.js and so on and so on till we find the correct default value.